staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -93,15 +93,40 @@ type authProxyRoundTripper struct {
 	rt http.RoundTripper
 }
 
+// defaultNoStoreRoundTripper is a round-tripper that sets the default
+// Cache-control to "no-store, no-cache" if no value has been set. This
+// will prevent us from caching zealously.
+type defaultNoStoreRoundTripper struct {
+	rt *httpcache.Transport
+}
+
+func (t *defaultNoStoreRoundTripper) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	cacheControl := req.Header.Get("Cache-control")
+
+	// t.rt shouldn't cache if CacheControl is set to no-store, or
+	// no-cache, but let's be extra careful.
+	if strings.Contains(cacheControl, "no-store") ||
+		strings.Contains(cacheControl, "no-cache") ||
+		cacheControl == "" {
+		// Bypass cache layer.
+		return t.rt.Transport.RoundTrip(req)
+	}
+	return t.rt.RoundTrip(req)
```

Comment:
> could they get each other's content

Do you mean could they see the cache-control header set by the other round tripper?

I suspect that they could.  @apelisse do you know.